src/bucket/BucketManagerImpl.cpp

```diff
@@ -960,6 +1136,26 @@ BucketManagerImpl::getBloomLookupMeter() const
     return mBucketListDBBloomLookups;
 }
 
+uint64_t
+BucketManagerImpl::getAverageBucketListSize(AbstractLedgerTxn& ltx) const
```

Comment:
This probably should belong to the `NetworkConfig` in order to have all the lazy loading logic in one spot (and then we could hopefully get rid of lazy loading altogether). 
 think most of the code here should actually be there, besides the bits that actually compute the bucket list size. Is there any particular reason for having this in the bucket list manager?